packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -380,8 +380,10 @@ abstract class ImplicitlyAnimatedWidgetState<T extends ImplicitlyAnimatedWidget>
   @override
   void didUpdateWidget(T oldWidget) {
     super.didUpdateWidget(oldWidget);
-    if (widget.curve != oldWidget.curve)
+    if (widget.curve != oldWidget.curve) {
+      (_animation as CurvedAnimation).dispose();
```

Comment:
We should probably add this to `dispose` below too, right?

Reply:
Also - can you add a test for the widget as well?

Reply:
I don't think you need to dispose in dispose as it already disposes the controller which cleans up the status listener.

Reply:
I was thinking about it but I have no idea how I can verify that we're calling dispose here.  Do you have a suggestion?

Reply:
Fish out the `animation` from the state before updating the widget, update the widget, and then verify that changing the status on the controller doesn't change the animation.

Reply:
Maybe just adding a comment in dispose about how disposing the parent takes care of the child would be good then to avoid confusion for future us :)